src/shared/loghandler.cpp

```diff
@@ -70,6 +87,14 @@ void LogHandler::messageHandler(LogLevel logLevel, const QString& className,
   maybeCreate(lock)->addLog(Log(logLevel, className, message), lock);
 }
 
+// static
+void LogHandler::rustMessageHandler(int32_t logLevel, char* message) {
+  MutexLocker lock(&s_mutex);
+
+  maybeCreate(lock)->addLog(
+      Log(intToLogLevel(logLevel), "Rust", QString::fromUtf8(message)), lock);
```

Comment:
Ideally this would not just say "Rust", but actually include the crate name. That added a bit more complexity than I was willing to endure for this ticket. Also, vpnglean is the only Rust crate as of right now that is sending logs to the VPN log stream anyways. If we decide to also send Signature logs we will probably need to reasses.

Reply:
Can you file a Jira issue for this?

Reply:
done: https://mozilla-hub.atlassian.net/browse/VPN-3668